app/models/document.rb

```diff
@@ -1,27 +1,14 @@
-class AbstractDocument
+class Document
   attr_reader :title
 
-  def self.date_metadata_keys
-    []
-  end
-
-  def self.tag_metadata_keys
-    []
-  end
-
-  def self.metadata_keys
-    tag_metadata_keys + date_metadata_keys
-  end
-
-  def self.metadata_name_mappings
-    {}
-  end
-
-  def initialize(attrs)
+  def initialize(attrs, finder)
+    attrs = attrs.with_indifferent_access
```

Comment:
We should make the test data reflect the real world data – otherwise someone could remove this line and replace all the `fetch` arguments with strings, and the tests would indicate everything is fine